### PR TITLE
Avoid some in-code ifdefs in reactor.cc

### PR DESCRIPTION
**Pull request by xemul**
Preprocessor conditionals (#if, #ifdef) in the middle of a function body make code harder to read and logic harder to follow. This patch re-ifdefs some of them to produce specific or stub implementations for different if/else cases. The compiler is good at not generating any code for stub calls.

### CODE REVIEW

File: src/core/reactor.cc [hunk 1]
```diff
@@ -891,6 +892,13 @@ static decltype(auto) install_signal_handler_stack() {
         }
     });
 }
+#else
+// SIGSTKSZ is too small when using asan. We also don't need to
+// handle SIGSEGV ourselves when using asan, so just don't install
+// a signal handler stack.
+struct nothing { ~nothing() {} };
+nothing install_signal_handler_stack() { return nothing{}; }
```
Comment (tchaikov) [inline src/core/reactor.cc:900]: nit, could move `nothing` into the body of `install_signal_handler_stack()`, and make the function to return `auto`, so the symbol of `nothing` is not exposed, and we don't need to move it into an anonymous namespace.

Comment (xemul) [reply src/core/reactor.cc:900]: Hm, good point, will try

Comment (xemul) [reply src/core/reactor.cc]: Yup, it did the trick, thanks!

**Comment by xemul** [comment]
Upd:
- moved the `struct nothing` stub into a function scope not to pollute file namespace

**Review by tchaikov — APPROVED** [review]
lgtm

**Pull request merged**